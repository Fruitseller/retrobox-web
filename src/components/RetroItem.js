import React from 'react';
import PropTypes from 'prop-types';
import {
  ListItem,
  ListItemText,
  ListItemSecondaryAction
} from 'material-ui/List';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from 'material-ui-icons/Delete';
import Divider from 'material-ui/Divider';

const RetroItem = props => {
  const {
    teamId,
    authorId,
    authorName,
    timestamp,
    message,
    removeItem
  } = props;
  return (
    <div>
      <Divider />
      <ListItem key={timestamp}>
        <ListItemText primary={message} secondary={authorName || 'Anonymous'} />
        <ListItemSecondaryAction>
          <IconButton
            aria-label="Delete"
            onClick={() => removeItem(teamId, authorId, timestamp)}
          >
            <DeleteIcon />
          </IconButton>
        </ListItemSecondaryAction>
      </ListItem>
    </div>
  );
};

RetroItem.propTypes = {
  teamId: PropTypes.string.isRequired,
  authorId: PropTypes.string.isRequired,
  authorName: PropTypes.string,
  timestamp: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  removeItem: PropTypes.func.isRequired
};

export default RetroItem;
