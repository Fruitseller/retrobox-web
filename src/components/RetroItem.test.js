import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import RetroItem from './RetroItem';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  shallow(
    <RetroItem
      teamId="teamId"
      authorId="authorId"
      authorName="displayName"
      timestamp={12345678}
      message="massage"
      removeItem={() => {}}
    />
  );
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <RetroItem
        teamId="teamId"
        authorId="authorId"
        authorName="displayName"
        timestamp={12345678}
        message="massage"
        removeItem={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
