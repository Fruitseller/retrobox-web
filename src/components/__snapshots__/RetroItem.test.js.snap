// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <hr
    className="MuiDivider-root-1 MuiDivider-default-2"
  />
  <div
    className="MuiListItem-container-7"
  >
    <li
      className="MuiListItem-root-6 MuiListItem-gutters-13 MuiListItem-secondaryAction-15 MuiListItem-default-9"
      disabled={false}
    >
      <div
        className="MuiListItemText-root-16"
      >
        <h3
          className="MuiTypography-root-21 MuiTypography-subheading-28 MuiListItemText-text-19"
        >
          massage
        </h3>
        <p
          className="MuiTypography-root-21 MuiTypography-body1-30 MuiTypography-colorSecondary-42 MuiListItemText-text-19"
        >
          displayName
        </p>
      </div>
    </li>
    <div
      className="MuiListItemSecondaryAction-root-45"
    >
      <button
        aria-label="Delete"
        className="MuiButtonBase-root-55 MuiIconButton-root-46"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label-52"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-57 MuiIconButton-icon-53"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-58"
        />
      </button>
    </div>
  </div>
</div>
`;
